platforms/core-configuration/model-core/src/main/java/org/gradle/model/internal/core/TypeCompatibilityModelProjectionSupport.java

```diff
@@ -45,7 +45,7 @@ public <T> boolean canBeViewedAs(ModelType<T> targetType) {
 
     private <T> boolean canBeAssignedTo(ModelType<T> targetType) {
         return targetType.isAssignableFrom(type)
-            || (targetType == ModelType.UNTYPED && isPrimitiveType(type));
+            || (ModelType.UNTYPED.equals(targetType) && isPrimitiveType(type));
```

Comment:
💭 This seems to be a safe change, most other places do check `UNTYPED` with `equals`.